libafl/src/fuzzer/mod.rs

```diff
@@ -612,14 +652,38 @@ where
         #[cfg(feature = "introspection")]
         state.introspection_monitor_mut().mark_manager_time();
 
+        let mut fuzzed_flag = false;
         {
             let mut testcase = state.testcase_mut(idx)?;
             let scheduled_count = testcase.scheduled_count();
-
+            if scheduled_count == 0 {
+                // Mark this testcase is fuzzed
+                fuzzed_flag = true;
+            }
             // increase scheduled count, this was fuzz_level in afl
             testcase.set_scheduled_count(scheduled_count + 1);
         }
 
+        if fuzzed_flag {
+            let pending_size = state.pending_mut();
+            if *pending_size > 0 {
+                *pending_size -= 1;
+            }
+        }
+
+        let pending_size = *state.pending();
+        let own_finds_size = *state.own_finds();
+        let imported_size = *state.imported();
+
+        manager.fire(
```

Comment:
I don't think we want to do this at every fuzz_one round 
can you do this in `fn maybe_report_progress`?

Reply:
yes I agree